OpenRA.Game/Graphics/Sheet.cs

```diff
@@ -106,12 +106,16 @@ public Png AsPng(TextureChannel channel, IPalette pal)
 			return new Png(plane, SpriteFrameType.Indexed8, Size.Width, Size.Height, palColors);
 		}
 
-		public void CreateBuffer()
+		public void CreateBuffer(byte backgroundPaletteIndex)
 		{
 			if (data != null)
 				return;
 			if (texture == null)
+			{
 				data = new byte[4 * Size.Width * Size.Height];
+				if (backgroundPaletteIndex != 0)
```

Comment:
This applies to all sheets, not just indexed ones. This moves the problem from indexed sheets over to 32 bit ones!